{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## <font color=\"green\"> Human Protein Atlas </font>\n",
    "<font color=\"blue\"> Read the data from the HPA project </font>"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "import numpy as np\n",
    "import pandas as pd\n",
    "from Bio import SeqIO\n",
    "import csv"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "<class 'http.client.HTTPResponse'>\n",
      "<class 'bytes'>\n"
     ]
    }
   ],
   "source": [
    "#import urllib2\n",
    "import urllib.request\n",
    "fhand = urllib.request.urlopen('https://www.proteinatlas.org/download/normal_tissue.tsv.zip')\n",
    "#response = urllib2.urlopen('https://www.proteinatlas.org/download/normal_tissue.tsv.zip')\n",
    "zipcontent= fhand.read()\n",
    "print(type(fhand))\n",
    "print(type(zipcontent))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(1056061, 6)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene</th>\n",
       "      <th>Gene name</th>\n",
       "      <th>Tissue</th>\n",
       "      <th>Cell type</th>\n",
       "      <th>Level</th>\n",
       "      <th>Reliability</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>ENSG00000000003</td>\n",
       "      <td>TSPAN6</td>\n",
       "      <td>adipose tissue</td>\n",
       "      <td>adipocytes</td>\n",
       "      <td>Not detected</td>\n",
       "      <td>Approved</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>ENSG00000000003</td>\n",
       "      <td>TSPAN6</td>\n",
       "      <td>adrenal gland</td>\n",
       "      <td>glandular cells</td>\n",
       "      <td>Not detected</td>\n",
       "      <td>Approved</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2</td>\n",
       "      <td>ENSG00000000003</td>\n",
       "      <td>TSPAN6</td>\n",
       "      <td>appendix</td>\n",
       "      <td>glandular cells</td>\n",
       "      <td>Medium</td>\n",
       "      <td>Approved</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>3</td>\n",
       "      <td>ENSG00000000003</td>\n",
       "      <td>TSPAN6</td>\n",
       "      <td>appendix</td>\n",
       "      <td>lymphoid tissue</td>\n",
       "      <td>Not detected</td>\n",
       "      <td>Approved</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>4</td>\n",
       "      <td>ENSG00000000003</td>\n",
       "      <td>TSPAN6</td>\n",
       "      <td>bone marrow</td>\n",
       "      <td>hematopoietic cells</td>\n",
       "      <td>Not detected</td>\n",
       "      <td>Approved</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "              Gene Gene name          Tissue            Cell type  \\\n",
       "0  ENSG00000000003    TSPAN6  adipose tissue           adipocytes   \n",
       "1  ENSG00000000003    TSPAN6   adrenal gland      glandular cells   \n",
       "2  ENSG00000000003    TSPAN6        appendix      glandular cells   \n",
       "3  ENSG00000000003    TSPAN6        appendix      lymphoid tissue   \n",
       "4  ENSG00000000003    TSPAN6     bone marrow  hematopoietic cells   \n",
       "\n",
       "          Level Reliability  \n",
       "0  Not detected    Approved  \n",
       "1  Not detected    Approved  \n",
       "2        Medium    Approved  \n",
       "3  Not detected    Approved  \n",
       "4  Not detected    Approved  "
      ]
     },
     "execution_count": 3,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "import requests, zipfile, io\n",
    "r = requests.get('https://www.proteinatlas.org/download/normal_tissue.tsv.zip')\n",
    "z = zipfile.ZipFile(io.BytesIO(r.content))\n",
    "z.extractall(\"/Users/saheeba/Downloads/\")\n",
    "#filepath= \"/Users/saheeba/Downloads/Human_atlas_subcellular_location.tsv\"\n",
    "HP_atlas = pd.read_csv(z.open('normal_tissue.tsv'), sep = \"\\t\")\n",
    "print(HP_atlas.shape)\n",
    "HP_atlas.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "975\n",
      "Done!\n"
     ]
    }
   ],
   "source": [
    "bingo_new = pd.read_csv('genesofinterest_human_atlas.csv')\n",
    "genes = bingo_new.iloc[:,0].tolist()\n",
    "print(len(genes))\n",
    "print(\"Done!\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(48993, 6)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene</th>\n",
       "      <th>Gene name</th>\n",
       "      <th>Tissue</th>\n",
       "      <th>Cell type</th>\n",
       "      <th>Level</th>\n",
       "      <th>Reliability</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>865</td>\n",
       "      <td>ENSG00000001561</td>\n",
       "      <td>ENPP4</td>\n",
       "      <td>adipose tissue</td>\n",
       "      <td>adipocytes</td>\n",
       "      <td>Low</td>\n",
       "      <td>Approved</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>866</td>\n",
       "      <td>ENSG00000001561</td>\n",
       "      <td>ENPP4</td>\n",
       "      <td>adrenal gland</td>\n",
       "      <td>glandular cells</td>\n",
       "      <td>High</td>\n",
       "      <td>Approved</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>867</td>\n",
       "      <td>ENSG00000001561</td>\n",
       "      <td>ENPP4</td>\n",
       "      <td>appendix</td>\n",
       "      <td>glandular cells</td>\n",
       "      <td>Medium</td>\n",
       "      <td>Approved</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>868</td>\n",
       "      <td>ENSG00000001561</td>\n",
       "      <td>ENPP4</td>\n",
       "      <td>appendix</td>\n",
       "      <td>lymphoid tissue</td>\n",
       "      <td>Low</td>\n",
       "      <td>Approved</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>869</td>\n",
       "      <td>ENSG00000001561</td>\n",
       "      <td>ENPP4</td>\n",
       "      <td>bone marrow</td>\n",
       "      <td>hematopoietic cells</td>\n",
       "      <td>Medium</td>\n",
       "      <td>Approved</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                Gene Gene name          Tissue            Cell type   Level  \\\n",
       "865  ENSG00000001561     ENPP4  adipose tissue           adipocytes     Low   \n",
       "866  ENSG00000001561     ENPP4   adrenal gland      glandular cells    High   \n",
       "867  ENSG00000001561     ENPP4        appendix      glandular cells  Medium   \n",
       "868  ENSG00000001561     ENPP4        appendix      lymphoid tissue     Low   \n",
       "869  ENSG00000001561     ENPP4     bone marrow  hematopoietic cells  Medium   \n",
       "\n",
       "    Reliability  \n",
       "865    Approved  \n",
       "866    Approved  \n",
       "867    Approved  \n",
       "868    Approved  \n",
       "869    Approved  "
      ]
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "protein_atlas = HP_atlas.loc[HP_atlas['Gene name'].isin(genes)]\n",
    "print(protein_atlas.shape)\n",
    "export_csv = protein_atlas.to_csv (r'/Users/saheeba/Downloads/Corresponding_human_atlas.csv', index = None, header=True)\n",
    "protein_atlas.head()\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(48993, 6)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene</th>\n",
       "      <th>Gene name</th>\n",
       "      <th>Tissue</th>\n",
       "      <th>Cell type</th>\n",
       "      <th>Level</th>\n",
       "      <th>Reliability</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>629557</td>\n",
       "      <td>ENSG00000159640</td>\n",
       "      <td>ACE</td>\n",
       "      <td>soft tissue 1</td>\n",
       "      <td>fibroblasts</td>\n",
       "      <td>Not detected</td>\n",
       "      <td>Enhanced</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>629514</td>\n",
       "      <td>ENSG00000159640</td>\n",
       "      <td>ACE</td>\n",
       "      <td>colon</td>\n",
       "      <td>glandular cells</td>\n",
       "      <td>Not detected</td>\n",
       "      <td>Enhanced</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>629515</td>\n",
       "      <td>ENSG00000159640</td>\n",
       "      <td>ACE</td>\n",
       "      <td>colon</td>\n",
       "      <td>peripheral nerve/ganglion</td>\n",
       "      <td>Not detected</td>\n",
       "      <td>Enhanced</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                   Gene Gene name         Tissue                  Cell type  \\\n",
       "629557  ENSG00000159640       ACE  soft tissue 1                fibroblasts   \n",
       "629514  ENSG00000159640       ACE          colon            glandular cells   \n",
       "629515  ENSG00000159640       ACE          colon  peripheral nerve/ganglion   \n",
       "\n",
       "               Level Reliability  \n",
       "629557  Not detected    Enhanced  \n",
       "629514  Not detected    Enhanced  \n",
       "629515  Not detected    Enhanced  "
      ]
     },
     "execution_count": 6,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "protein_atlas.loc[protein_atlas['Gene name'].isin(genes)]\n",
    "sorted_atlas = protein_atlas.sort_values(by=['Gene name'], ascending=True)\n",
    "print(sorted_atlas.shape)\n",
    "sorted_atlas.head(3)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### <font color=\"blue\">Conditional combine all culoumns and stack them together | dataframes"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>positionID</th>\n",
       "      <th>Entry</th>\n",
       "      <th>Entry name</th>\n",
       "      <th>Status</th>\n",
       "      <th>Protein names</th>\n",
       "      <th>Gene names</th>\n",
       "      <th>Organism</th>\n",
       "      <th>Length</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>172</td>\n",
       "      <td>P12821[30-1256]</td>\n",
       "      <td>P12821</td>\n",
       "      <td>ACE_HUMAN</td>\n",
       "      <td>reviewed</td>\n",
       "      <td>Angiotensin-converting enzyme (ACE) (EC 3.2.1....</td>\n",
       "      <td>ACE DCP DCP1</td>\n",
       "      <td>Homo sapiens (Human)</td>\n",
       "      <td>1306</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>794</td>\n",
       "      <td>Q9BYF1[18-740]</td>\n",
       "      <td>Q9BYF1</td>\n",
       "      <td>ACE2_HUMAN</td>\n",
       "      <td>reviewed</td>\n",
       "      <td>Angiotensin-converting enzyme 2 (EC 3.4.17.23)...</td>\n",
       "      <td>ACE2 UNQ868/PRO1885</td>\n",
       "      <td>Homo sapiens (Human)</td>\n",
       "      <td>805</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>273</td>\n",
       "      <td>Q04771[21-123]</td>\n",
       "      <td>Q04771</td>\n",
       "      <td>ACVR1_HUMAN</td>\n",
       "      <td>reviewed</td>\n",
       "      <td>Activin receptor type-1 (EC 2.7.11.30) (Activi...</td>\n",
       "      <td>ACVR1 ACVRLK2</td>\n",
       "      <td>Homo sapiens (Human)</td>\n",
       "      <td>509</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>931</td>\n",
       "      <td>D3DPA4[1-145]</td>\n",
       "      <td>D3DPA4</td>\n",
       "      <td>D3DPA4_HUMAN</td>\n",
       "      <td>unreviewed</td>\n",
       "      <td>Serine/threonine-protein kinase receptor (EC 2...</td>\n",
       "      <td>ACVR1 hCG_1811747</td>\n",
       "      <td>Homo sapiens (Human)</td>\n",
       "      <td>509</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>87</td>\n",
       "      <td>P36896[24-126]</td>\n",
       "      <td>P36896</td>\n",
       "      <td>ACV1B_HUMAN</td>\n",
       "      <td>reviewed</td>\n",
       "      <td>Activin receptor type-1B (EC 2.7.11.30) (Activ...</td>\n",
       "      <td>ACVR1B ACVRLK4 ALK4</td>\n",
       "      <td>Homo sapiens (Human)</td>\n",
       "      <td>505</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "          positionID   Entry    Entry name      Status  \\\n",
       "172  P12821[30-1256]  P12821     ACE_HUMAN    reviewed   \n",
       "794   Q9BYF1[18-740]  Q9BYF1    ACE2_HUMAN    reviewed   \n",
       "273   Q04771[21-123]  Q04771   ACVR1_HUMAN    reviewed   \n",
       "931    D3DPA4[1-145]  D3DPA4  D3DPA4_HUMAN  unreviewed   \n",
       "87    P36896[24-126]  P36896   ACV1B_HUMAN    reviewed   \n",
       "\n",
       "                                         Protein names           Gene names  \\\n",
       "172  Angiotensin-converting enzyme (ACE) (EC 3.2.1....         ACE DCP DCP1   \n",
       "794  Angiotensin-converting enzyme 2 (EC 3.4.17.23)...  ACE2 UNQ868/PRO1885   \n",
       "273  Activin receptor type-1 (EC 2.7.11.30) (Activi...        ACVR1 ACVRLK2   \n",
       "931  Serine/threonine-protein kinase receptor (EC 2...    ACVR1 hCG_1811747   \n",
       "87   Activin receptor type-1B (EC 2.7.11.30) (Activ...  ACVR1B ACVRLK4 ALK4   \n",
       "\n",
       "                 Organism  Length  \n",
       "172  Homo sapiens (Human)    1306  \n",
       "794  Homo sapiens (Human)     805  \n",
       "273  Homo sapiens (Human)     509  \n",
       "931  Homo sapiens (Human)     509  \n",
       "87   Homo sapiens (Human)     505  "
      ]
     },
     "execution_count": 7,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "custom_seqs = pd.read_excel('customisedMappingID.xlsx')\n",
    "sorted_custom_seq = custom_seqs.sort_values('Gene names',ascending=True)\n",
    "sorted_custom_seq.rename(columns={'yourlist:M202001296746803381A1F0E0DB47453E0216320D6E4B519':'positionID'},inplace=True)\n",
    "sorted_custom_seq.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(619, 5)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th>Level</th>\n",
       "      <th>Gene name</th>\n",
       "      <th>High</th>\n",
       "      <th>Low</th>\n",
       "      <th>Medium</th>\n",
       "      <th>Not detected</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>ACE</td>\n",
       "      <td>glandular cells, glandular cells, glandular ce...</td>\n",
       "      <td>endothelial cells</td>\n",
       "      <td>glandular cells, pneumocytes, cells in seminif...</td>\n",
       "      <td>fibroblasts, glandular cells, peripheral nerve...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>ACE2</td>\n",
       "      <td>glandular cells, Leydig cells, cells in semini...</td>\n",
       "      <td>glandular cells, glandular cells, glandular ce...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>adipocytes, smooth muscle cells, epidermal cel...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2</td>\n",
       "      <td>ACVR1</td>\n",
       "      <td>glandular cells, smooth muscle cells</td>\n",
       "      <td>squamous epithelial cells, glial cells, pneumo...</td>\n",
       "      <td>myocytes, glandular cells, glandular cells, gl...</td>\n",
       "      <td>cells in endometrial stroma, glandular cells, ...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "Level Gene name                                               High  \\\n",
       "0           ACE  glandular cells, glandular cells, glandular ce...   \n",
       "1          ACE2  glandular cells, Leydig cells, cells in semini...   \n",
       "2         ACVR1               glandular cells, smooth muscle cells   \n",
       "\n",
       "Level                                                Low  \\\n",
       "0                                      endothelial cells   \n",
       "1      glandular cells, glandular cells, glandular ce...   \n",
       "2      squamous epithelial cells, glial cells, pneumo...   \n",
       "\n",
       "Level                                             Medium  \\\n",
       "0      glandular cells, pneumocytes, cells in seminif...   \n",
       "1                                                    NaN   \n",
       "2      myocytes, glandular cells, glandular cells, gl...   \n",
       "\n",
       "Level                                       Not detected  \n",
       "0      fibroblasts, glandular cells, peripheral nerve...  \n",
       "1      adipocytes, smooth muscle cells, epidermal cel...  \n",
       "2      cells in endometrial stroma, glandular cells, ...  "
      ]
     },
     "execution_count": 8,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df = sorted_atlas.groupby(['Gene name','Level'], as_index=False)['Cell type'].agg(', '.join).set_index(['Gene name','Level'])['Cell type'].unstack().reset_index()\n",
    "print(df.shape) \n",
    "df.head(3)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(645, 13)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>positionID</th>\n",
       "      <th>Entry</th>\n",
       "      <th>Entry name</th>\n",
       "      <th>Status</th>\n",
       "      <th>Protein names</th>\n",
       "      <th>Gene names</th>\n",
       "      <th>Organism</th>\n",
       "      <th>Length</th>\n",
       "      <th>Gene name</th>\n",
       "      <th>High</th>\n",
       "      <th>Low</th>\n",
       "      <th>Medium</th>\n",
       "      <th>Not detected</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>P12821[30-1256]</td>\n",
       "      <td>P12821</td>\n",
       "      <td>ACE_HUMAN</td>\n",
       "      <td>reviewed</td>\n",
       "      <td>Angiotensin-converting enzyme (ACE) (EC 3.2.1....</td>\n",
       "      <td>ACE DCP DCP1</td>\n",
       "      <td>Homo sapiens (Human)</td>\n",
       "      <td>1306</td>\n",
       "      <td>ACE</td>\n",
       "      <td>glandular cells, glandular cells, glandular ce...</td>\n",
       "      <td>endothelial cells</td>\n",
       "      <td>glandular cells, pneumocytes, cells in seminif...</td>\n",
       "      <td>fibroblasts, glandular cells, peripheral nerve...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>Q9BYF1[18-740]</td>\n",
       "      <td>Q9BYF1</td>\n",
       "      <td>ACE2_HUMAN</td>\n",
       "      <td>reviewed</td>\n",
       "      <td>Angiotensin-converting enzyme 2 (EC 3.4.17.23)...</td>\n",
       "      <td>ACE2 UNQ868/PRO1885</td>\n",
       "      <td>Homo sapiens (Human)</td>\n",
       "      <td>805</td>\n",
       "      <td>ACE2</td>\n",
       "      <td>glandular cells, Leydig cells, cells in semini...</td>\n",
       "      <td>glandular cells, glandular cells, glandular ce...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>adipocytes, smooth muscle cells, epidermal cel...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2</td>\n",
       "      <td>Q04771[21-123]</td>\n",
       "      <td>Q04771</td>\n",
       "      <td>ACVR1_HUMAN</td>\n",
       "      <td>reviewed</td>\n",
       "      <td>Activin receptor type-1 (EC 2.7.11.30) (Activi...</td>\n",
       "      <td>ACVR1 ACVRLK2</td>\n",
       "      <td>Homo sapiens (Human)</td>\n",
       "      <td>509</td>\n",
       "      <td>ACVR1</td>\n",
       "      <td>glandular cells, smooth muscle cells</td>\n",
       "      <td>squamous epithelial cells, glial cells, pneumo...</td>\n",
       "      <td>myocytes, glandular cells, glandular cells, gl...</td>\n",
       "      <td>cells in endometrial stroma, glandular cells, ...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "        positionID   Entry   Entry name    Status  \\\n",
       "0  P12821[30-1256]  P12821    ACE_HUMAN  reviewed   \n",
       "1   Q9BYF1[18-740]  Q9BYF1   ACE2_HUMAN  reviewed   \n",
       "2   Q04771[21-123]  Q04771  ACVR1_HUMAN  reviewed   \n",
       "\n",
       "                                       Protein names           Gene names  \\\n",
       "0  Angiotensin-converting enzyme (ACE) (EC 3.2.1....         ACE DCP DCP1   \n",
       "1  Angiotensin-converting enzyme 2 (EC 3.4.17.23)...  ACE2 UNQ868/PRO1885   \n",
       "2  Activin receptor type-1 (EC 2.7.11.30) (Activi...        ACVR1 ACVRLK2   \n",
       "\n",
       "               Organism  Length Gene name  \\\n",
       "0  Homo sapiens (Human)    1306       ACE   \n",
       "1  Homo sapiens (Human)     805      ACE2   \n",
       "2  Homo sapiens (Human)     509     ACVR1   \n",
       "\n",
       "                                                High  \\\n",
       "0  glandular cells, glandular cells, glandular ce...   \n",
       "1  glandular cells, Leydig cells, cells in semini...   \n",
       "2               glandular cells, smooth muscle cells   \n",
       "\n",
       "                                                 Low  \\\n",
       "0                                  endothelial cells   \n",
       "1  glandular cells, glandular cells, glandular ce...   \n",
       "2  squamous epithelial cells, glial cells, pneumo...   \n",
       "\n",
       "                                              Medium  \\\n",
       "0  glandular cells, pneumocytes, cells in seminif...   \n",
       "1                                                NaN   \n",
       "2  myocytes, glandular cells, glandular cells, gl...   \n",
       "\n",
       "                                        Not detected  \n",
       "0  fibroblasts, glandular cells, peripheral nerve...  \n",
       "1  adipocytes, smooth muscle cells, epidermal cel...  \n",
       "2  cells in endometrial stroma, glandular cells, ...  "
      ]
     },
     "execution_count": 9,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "sorted_custom_seq['Gene name'] = genes\n",
    "s1 = pd.merge(sorted_custom_seq,df ,how='inner', on=['Gene name'])\n",
    "export = s1.to_csv('merged_expression.csv',index = None, header=True)\n",
    "print(s1.shape)\n",
    "s1.head(3)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>positionID</th>\n",
       "      <th>Entry</th>\n",
       "      <th>Protein names</th>\n",
       "      <th>Gene names</th>\n",
       "      <th>Length</th>\n",
       "      <th>Protein Expression : High</th>\n",
       "      <th>Protein Expression : Low</th>\n",
       "      <th>Protein Expression : Med</th>\n",
       "      <th>Not detected</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>P12821[30-1256]</td>\n",
       "      <td>P12821</td>\n",
       "      <td>Angiotensin-converting enzyme (ACE) (EC 3.2.1....</td>\n",
       "      <td>ACE DCP DCP1</td>\n",
       "      <td>1306</td>\n",
       "      <td>glandular cells, glandular cells, glandular ce...</td>\n",
       "      <td>endothelial cells</td>\n",
       "      <td>glandular cells, pneumocytes, cells in seminif...</td>\n",
       "      <td>fibroblasts, glandular cells, peripheral nerve...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>Q9BYF1[18-740]</td>\n",
       "      <td>Q9BYF1</td>\n",
       "      <td>Angiotensin-converting enzyme 2 (EC 3.4.17.23)...</td>\n",
       "      <td>ACE2 UNQ868/PRO1885</td>\n",
       "      <td>805</td>\n",
       "      <td>glandular cells, Leydig cells, cells in semini...</td>\n",
       "      <td>glandular cells, glandular cells, glandular ce...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>adipocytes, smooth muscle cells, epidermal cel...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2</td>\n",
       "      <td>Q04771[21-123]</td>\n",
       "      <td>Q04771</td>\n",
       "      <td>Activin receptor type-1 (EC 2.7.11.30) (Activi...</td>\n",
       "      <td>ACVR1 ACVRLK2</td>\n",
       "      <td>509</td>\n",
       "      <td>glandular cells, smooth muscle cells</td>\n",
       "      <td>squamous epithelial cells, glial cells, pneumo...</td>\n",
       "      <td>myocytes, glandular cells, glandular cells, gl...</td>\n",
       "      <td>cells in endometrial stroma, glandular cells, ...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>3</td>\n",
       "      <td>D3DPA4[1-145]</td>\n",
       "      <td>D3DPA4</td>\n",
       "      <td>Serine/threonine-protein kinase receptor (EC 2...</td>\n",
       "      <td>ACVR1 hCG_1811747</td>\n",
       "      <td>509</td>\n",
       "      <td>glandular cells, smooth muscle cells</td>\n",
       "      <td>squamous epithelial cells, glial cells, pneumo...</td>\n",
       "      <td>myocytes, glandular cells, glandular cells, gl...</td>\n",
       "      <td>cells in endometrial stroma, glandular cells, ...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>4</td>\n",
       "      <td>P36896[24-126]</td>\n",
       "      <td>P36896</td>\n",
       "      <td>Activin receptor type-1B (EC 2.7.11.30) (Activ...</td>\n",
       "      <td>ACVR1B ACVRLK4 ALK4</td>\n",
       "      <td>505</td>\n",
       "      <td>NaN</td>\n",
       "      <td>glandular cells, decidual cells, squamous epit...</td>\n",
       "      <td>exocrine glandular cells, glandular cells, gla...</td>\n",
       "      <td>melanocytes, Langerhans, keratinocytes, fibrob...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "        positionID   Entry                                      Protein names  \\\n",
       "0  P12821[30-1256]  P12821  Angiotensin-converting enzyme (ACE) (EC 3.2.1....   \n",
       "1   Q9BYF1[18-740]  Q9BYF1  Angiotensin-converting enzyme 2 (EC 3.4.17.23)...   \n",
       "2   Q04771[21-123]  Q04771  Activin receptor type-1 (EC 2.7.11.30) (Activi...   \n",
       "3    D3DPA4[1-145]  D3DPA4  Serine/threonine-protein kinase receptor (EC 2...   \n",
       "4   P36896[24-126]  P36896  Activin receptor type-1B (EC 2.7.11.30) (Activ...   \n",
       "\n",
       "            Gene names  Length  \\\n",
       "0         ACE DCP DCP1    1306   \n",
       "1  ACE2 UNQ868/PRO1885     805   \n",
       "2        ACVR1 ACVRLK2     509   \n",
       "3    ACVR1 hCG_1811747     509   \n",
       "4  ACVR1B ACVRLK4 ALK4     505   \n",
       "\n",
       "                           Protein Expression : High  \\\n",
       "0  glandular cells, glandular cells, glandular ce...   \n",
       "1  glandular cells, Leydig cells, cells in semini...   \n",
       "2               glandular cells, smooth muscle cells   \n",
       "3               glandular cells, smooth muscle cells   \n",
       "4                                                NaN   \n",
       "\n",
       "                            Protein Expression : Low  \\\n",
       "0                                  endothelial cells   \n",
       "1  glandular cells, glandular cells, glandular ce...   \n",
       "2  squamous epithelial cells, glial cells, pneumo...   \n",
       "3  squamous epithelial cells, glial cells, pneumo...   \n",
       "4  glandular cells, decidual cells, squamous epit...   \n",
       "\n",
       "                            Protein Expression : Med  \\\n",
       "0  glandular cells, pneumocytes, cells in seminif...   \n",
       "1                                                NaN   \n",
       "2  myocytes, glandular cells, glandular cells, gl...   \n",
       "3  myocytes, glandular cells, glandular cells, gl...   \n",
       "4  exocrine glandular cells, glandular cells, gla...   \n",
       "\n",
       "                                        Not detected  \n",
       "0  fibroblasts, glandular cells, peripheral nerve...  \n",
       "1  adipocytes, smooth muscle cells, epidermal cel...  \n",
       "2  cells in endometrial stroma, glandular cells, ...  \n",
       "3  cells in endometrial stroma, glandular cells, ...  \n",
       "4  melanocytes, Langerhans, keratinocytes, fibrob...  "
      ]
     },
     "execution_count": 10,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "s2 = s1.drop(['Status','Organism','Gene name','Entry name'], axis = 1)\n",
    "s2.rename(columns={'High':'Protein Expression : High','Low':'Protein Expression : Low','Medium':'Protein Expression : Med'},inplace=True)\n",
    "s2.head()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### <font color = 'blue'> To add corresponding sequences to the table </font>"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>ID</th>\n",
       "      <th>sequences</th>\n",
       "      <th>Seq_length</th>\n",
       "      <th>Entry</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>sp|Q9NR97|27-827</td>\n",
       "      <td>EENFSRSYPCDEKKQNDSVIAECSNRRLQEVPQTVGKYVTELDLSD...</td>\n",
       "      <td>801</td>\n",
       "      <td>Q9NR97</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>sp|Q02846|52-462</td>\n",
       "      <td>AVFTVGVLGPWACDPIFSRARPDLAARLAAARLNRDPGLAGGPRFE...</td>\n",
       "      <td>411</td>\n",
       "      <td>Q02846</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2</td>\n",
       "      <td>sp|Q9NY15|26-2478</td>\n",
       "      <td>QVLFKGCDVKTTFVTHVPCTSCAAIKKQTCPSGWLRELPDQITQDC...</td>\n",
       "      <td>2453</td>\n",
       "      <td>Q9NY15</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>3</td>\n",
       "      <td>sp|A6NFA1|20-494</td>\n",
       "      <td>RPQPPDGGQCRPPGSQRDLNSFLWTIRRDPPAYLFGTIHVPYTRVW...</td>\n",
       "      <td>475</td>\n",
       "      <td>A6NFA1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>4</td>\n",
       "      <td>sp|Q86V40|20-477</td>\n",
       "      <td>ASRRGAPGTANCELKPQQSELNSFLWTIKRDPPSYFFGTIHVPYTR...</td>\n",
       "      <td>458</td>\n",
       "      <td>Q86V40</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                  ID                                          sequences  \\\n",
       "0   sp|Q9NR97|27-827  EENFSRSYPCDEKKQNDSVIAECSNRRLQEVPQTVGKYVTELDLSD...   \n",
       "1   sp|Q02846|52-462  AVFTVGVLGPWACDPIFSRARPDLAARLAAARLNRDPGLAGGPRFE...   \n",
       "2  sp|Q9NY15|26-2478  QVLFKGCDVKTTFVTHVPCTSCAAIKKQTCPSGWLRELPDQITQDC...   \n",
       "3   sp|A6NFA1|20-494  RPQPPDGGQCRPPGSQRDLNSFLWTIRRDPPAYLFGTIHVPYTRVW...   \n",
       "4   sp|Q86V40|20-477  ASRRGAPGTANCELKPQQSELNSFLWTIKRDPPSYFFGTIHVPYTR...   \n",
       "\n",
       "   Seq_length   Entry  \n",
       "0         801  Q9NR97  \n",
       "1         411  Q02846  \n",
       "2        2453  Q9NY15  \n",
       "3         475  A6NFA1  \n",
       "4         458  Q86V40  "
      ]
     },
     "execution_count": 11,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "identifiers = []\n",
    "sequences = []\n",
    "lengths = []\n",
    "for seq_record in SeqIO.parse('/Users/saheeba/Downloads/splits_files/out.fasta', 'fasta'):  # (generator)\n",
    "        identifiers.append(seq_record.id)\n",
    "        sequences.append(str(seq_record.seq))#sequences.append(repr(seq_record.seq))#sequences.append(seq_record.seq)#.upper()\n",
    "        lengths.append(len(seq_record.seq))\n",
    "        s11 = pd.Series(identifiers, name='ID')\n",
    "        s12 = pd.Series(sequences, name='Seq')\n",
    "        s13 = pd.Series(lengths, name='len')\n",
    "        pdf = pd.DataFrame(dict(ID=s11, sequences=s12, Seq_length=s13))\n",
    "pdf['Entry']= pdf['ID'].str.split('|').str[1]\n",
    "pdf.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(645, 10)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>positionID</th>\n",
       "      <th>Entry</th>\n",
       "      <th>sequences</th>\n",
       "      <th>Protein names</th>\n",
       "      <th>Gene names</th>\n",
       "      <th>Seq_length</th>\n",
       "      <th>Protein Expression : High</th>\n",
       "      <th>Protein Expression : Low</th>\n",
       "      <th>Protein Expression : Med</th>\n",
       "      <th>Not detected</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>P12821[30-1256]</td>\n",
       "      <td>P12821</td>\n",
       "      <td>LDPGLQPGNFSADEAGAQLFAQSYNSSAEQVLFQSVAASWAHDTNI...</td>\n",
       "      <td>Angiotensin-converting enzyme (ACE) (EC 3.2.1....</td>\n",
       "      <td>ACE DCP DCP1</td>\n",
       "      <td>1227</td>\n",
       "      <td>glandular cells, glandular cells, glandular ce...</td>\n",
       "      <td>endothelial cells</td>\n",
       "      <td>glandular cells, pneumocytes, cells in seminif...</td>\n",
       "      <td>fibroblasts, glandular cells, peripheral nerve...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>Q9BYF1[18-740]</td>\n",
       "      <td>Q9BYF1</td>\n",
       "      <td>QSTIEEQAKTFLDKFNHEAEDLFYQSSLASWNYNTNITEENVQNMN...</td>\n",
       "      <td>Angiotensin-converting enzyme 2 (EC 3.4.17.23)...</td>\n",
       "      <td>ACE2 UNQ868/PRO1885</td>\n",
       "      <td>723</td>\n",
       "      <td>glandular cells, Leydig cells, cells in semini...</td>\n",
       "      <td>glandular cells, glandular cells, glandular ce...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>adipocytes, smooth muscle cells, epidermal cel...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2</td>\n",
       "      <td>Q04771[21-123]</td>\n",
       "      <td>Q04771</td>\n",
       "      <td>MEDEKPKVNPKLYMCVCEGLSCGNEDHCEGQQCFSSLSINDGFHVY...</td>\n",
       "      <td>Activin receptor type-1 (EC 2.7.11.30) (Activi...</td>\n",
       "      <td>ACVR1 ACVRLK2</td>\n",
       "      <td>103</td>\n",
       "      <td>glandular cells, smooth muscle cells</td>\n",
       "      <td>squamous epithelial cells, glial cells, pneumo...</td>\n",
       "      <td>myocytes, glandular cells, glandular cells, gl...</td>\n",
       "      <td>cells in endometrial stroma, glandular cells, ...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "        positionID   Entry                                          sequences  \\\n",
       "0  P12821[30-1256]  P12821  LDPGLQPGNFSADEAGAQLFAQSYNSSAEQVLFQSVAASWAHDTNI...   \n",
       "1   Q9BYF1[18-740]  Q9BYF1  QSTIEEQAKTFLDKFNHEAEDLFYQSSLASWNYNTNITEENVQNMN...   \n",
       "2   Q04771[21-123]  Q04771  MEDEKPKVNPKLYMCVCEGLSCGNEDHCEGQQCFSSLSINDGFHVY...   \n",
       "\n",
       "                                       Protein names           Gene names  \\\n",
       "0  Angiotensin-converting enzyme (ACE) (EC 3.2.1....         ACE DCP DCP1   \n",
       "1  Angiotensin-converting enzyme 2 (EC 3.4.17.23)...  ACE2 UNQ868/PRO1885   \n",
       "2  Activin receptor type-1 (EC 2.7.11.30) (Activi...        ACVR1 ACVRLK2   \n",
       "\n",
       "   Seq_length                          Protein Expression : High  \\\n",
       "0        1227  glandular cells, glandular cells, glandular ce...   \n",
       "1         723  glandular cells, Leydig cells, cells in semini...   \n",
       "2         103               glandular cells, smooth muscle cells   \n",
       "\n",
       "                            Protein Expression : Low  \\\n",
       "0                                  endothelial cells   \n",
       "1  glandular cells, glandular cells, glandular ce...   \n",
       "2  squamous epithelial cells, glial cells, pneumo...   \n",
       "\n",
       "                            Protein Expression : Med  \\\n",
       "0  glandular cells, pneumocytes, cells in seminif...   \n",
       "1                                                NaN   \n",
       "2  myocytes, glandular cells, glandular cells, gl...   \n",
       "\n",
       "                                        Not detected  \n",
       "0  fibroblasts, glandular cells, peripheral nerve...  \n",
       "1  adipocytes, smooth muscle cells, epidermal cel...  \n",
       "2  cells in endometrial stroma, glandular cells, ...  "
      ]
     },
     "execution_count": 12,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "s3 = pd.merge(s2,pdf ,how='inner', on=['Entry'])\n",
    "#s3.drop(['Status','Organism','Gene name','Entry name','Length','ID'], axis = 1)\n",
    "s3.head()\n",
    "#s3.rename(columns={'High':'Protein Expression : High','Low':'Protein Expression : Low','Medium':'Protein Expression : Medium'},inplace=True)\n",
    "s3 = s3[['positionID','Entry','sequences','Protein names','Gene names','Seq_length','Protein Expression : High','Protein Expression : Low','Protein Expression : Med','Not detected']]\n",
    "print(s3.shape)\n",
    "s3.head(3)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### <font color = 'blue'> Signaling pathway data: To add corresponding info to the table </font>"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(87581, 20)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>source_name</th>\n",
       "      <th>Source</th>\n",
       "      <th>source_speciesID</th>\n",
       "      <th>source_species</th>\n",
       "      <th>source_topology</th>\n",
       "      <th>source_pathways</th>\n",
       "      <th>target_name</th>\n",
       "      <th>Target</th>\n",
       "      <th>target_speciesID</th>\n",
       "      <th>target_species</th>\n",
       "      <th>target_topology</th>\n",
       "      <th>target_pathways</th>\n",
       "      <th>layer</th>\n",
       "      <th>interaction_type</th>\n",
       "      <th>directness</th>\n",
       "      <th>effect</th>\n",
       "      <th>references</th>\n",
       "      <th>source</th>\n",
       "      <th>confidence_score</th>\n",
       "      <th>score_from_the_source</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>NFKB1</td>\n",
       "      <td>P19838</td>\n",
       "      <td>ENSG00000109320</td>\n",
       "      <td>H. sapiens</td>\n",
       "      <td>Transcription factor</td>\n",
       "      <td>NaN</td>\n",
       "      <td>PRKCE</td>\n",
       "      <td>Q02156</td>\n",
       "      <td>ENSG00000171132</td>\n",
       "      <td>H. sapiens</td>\n",
       "      <td>Scaffold</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Transcriptional regulation</td>\n",
       "      <td>Transcriptional directed</td>\n",
       "      <td>indirect</td>\n",
       "      <td>unknown</td>\n",
       "      <td>14681366</td>\n",
       "      <td>JASPAR(url: http://jaspar.cgb.ki.se/ ,pmid: 14...</td>\n",
       "      <td>GO Semantic Similarity: 0.563594</td>\n",
       "      <td>Jaspar: 10.1804, Jaspar: 10.1804, Jaspar: 10.3...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>NFKB1</td>\n",
       "      <td>P19838</td>\n",
       "      <td>ENSG00000109320</td>\n",
       "      <td>H. sapiens</td>\n",
       "      <td>Transcription factor</td>\n",
       "      <td>NaN</td>\n",
       "      <td>PAG1</td>\n",
       "      <td>Q9NWQ8</td>\n",
       "      <td>ENSG00000076641</td>\n",
       "      <td>H. sapiens</td>\n",
       "      <td>Co-factor,Scaffold</td>\n",
       "      <td>RTK(non-core)</td>\n",
       "      <td>Transcriptional regulation</td>\n",
       "      <td>Transcriptional directed</td>\n",
       "      <td>indirect</td>\n",
       "      <td>unknown</td>\n",
       "      <td>14681366</td>\n",
       "      <td>JASPAR(url: http://jaspar.cgb.ki.se/ ,pmid: 14...</td>\n",
       "      <td>GO Semantic Similarity: 0.521359</td>\n",
       "      <td>Jaspar: 10.1804, Jaspar: 10.1804</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  source_name  Source source_speciesID source_species       source_topology  \\\n",
       "0       NFKB1  P19838  ENSG00000109320     H. sapiens  Transcription factor   \n",
       "1       NFKB1  P19838  ENSG00000109320     H. sapiens  Transcription factor   \n",
       "\n",
       "  source_pathways target_name  Target target_speciesID target_species  \\\n",
       "0             NaN       PRKCE  Q02156  ENSG00000171132     H. sapiens   \n",
       "1             NaN        PAG1  Q9NWQ8  ENSG00000076641     H. sapiens   \n",
       "\n",
       "      target_topology target_pathways                       layer  \\\n",
       "0            Scaffold             NaN  Transcriptional regulation   \n",
       "1  Co-factor,Scaffold   RTK(non-core)  Transcriptional regulation   \n",
       "\n",
       "           interaction_type directness   effect references  \\\n",
       "0  Transcriptional directed   indirect  unknown   14681366   \n",
       "1  Transcriptional directed   indirect  unknown   14681366   \n",
       "\n",
       "                                              source  \\\n",
       "0  JASPAR(url: http://jaspar.cgb.ki.se/ ,pmid: 14...   \n",
       "1  JASPAR(url: http://jaspar.cgb.ki.se/ ,pmid: 14...   \n",
       "\n",
       "                   confidence_score  \\\n",
       "0  GO Semantic Similarity: 0.563594   \n",
       "1  GO Semantic Similarity: 0.521359   \n",
       "\n",
       "                               score_from_the_source  \n",
       "0  Jaspar: 10.1804, Jaspar: 10.1804, Jaspar: 10.3...  \n",
       "1                   Jaspar: 10.1804, Jaspar: 10.1804  "
      ]
     },
     "execution_count": 29,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "SignaLink = pd.read_csv('signal_Book.csv')\n",
    "SignaLink.rename(columns={'source_uniprotAC':'Source','target_uniprotAC':'Target'},inplace=True)\n",
    "print(SignaLink.shape)\n",
    "SignaLink.head(2)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "\"SignaLink_1_0 = SignaLink.groupby(['Entry','target_pathways'], as_index=True)['Target'].agg(', '.join).unstack().reset_index()#.unstack().reset_index().set_index(['Entry','target_pathways'])['Target']\\nSignaLink_1_0.head(3)\""
      ]
     },
     "execution_count": 28,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "'''SignaLink_1_0 = SignaLink.groupby(['Entry','target_pathways'], as_index=True)['Target'].agg(', '.join).unstack().reset_index()#.unstack().reset_index().set_index(['Entry','target_pathways'])['Target']\n",
    "SignaLink_1_0.head(3)'''"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 59,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(87581, 2)\n",
      "(430,)\n",
      "(752,)\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "Target  target_pathways\n",
       "A7KAX9  RTK(non-core)      RTK(non-core),RTK(non-core),RTK(non-core),RTK(...\n",
       "A7MBM2  Hedgehog(core)     Hedgehog(core),Hedgehog(core),Hedgehog(core),H...\n",
       "C9JF09  RTK(non-core)                                          RTK(non-core)\n",
       "Name: target_pathways, dtype: object"
      ]
     },
     "execution_count": 59,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "SignaLink_source = SignaLink[['Source', 'source_pathways']]\n",
    "print(SignaLink_source.shape)\n",
    "source_match= SignaLink_source.groupby(['Source','source_pathways'], as_index=False)['source_pathways'].agg(\",\".join)\n",
    "print(source_match.shape)\n",
    "source_match.head(12)\n",
    "SignaLink_target = SignaLink[['Target', 'target_pathways']]\n",
    "target_match= SignaLink_target.groupby(['Target','target_pathways'], as_index=True)['target_pathways'].agg(\",\".join)\n",
    "#print(SignaLink_target.shape)\n",
    "print(target_match.shape)\n",
    "target_match.head(3)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(71, 53)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>positionID</th>\n",
       "      <th>Entry</th>\n",
       "      <th>Protein names</th>\n",
       "      <th>Gene names</th>\n",
       "      <th>Length</th>\n",
       "      <th>Protein Expression : High</th>\n",
       "      <th>Protein Expression : Low</th>\n",
       "      <th>Protein Expression : Med</th>\n",
       "      <th>Not detected</th>\n",
       "      <th>Hedgehog(core)</th>\n",
       "      <th>...</th>\n",
       "      <th>RTK(non-core),TGF(core)</th>\n",
       "      <th>RTK(non-core),TGF(non-core)</th>\n",
       "      <th>RTK(non-core),TGF(non-core),TGF(core),WNT/Wingless(core)</th>\n",
       "      <th>RTK(non-core),WNT/Wingless(non-core)</th>\n",
       "      <th>TGF(core)</th>\n",
       "      <th>TGF(non-core)</th>\n",
       "      <th>TGF(non-core),WNT/Wingless(core)</th>\n",
       "      <th>TGF(non-core),WNT/Wingless(non-core)</th>\n",
       "      <th>WNT/Wingless(core)</th>\n",
       "      <th>WNT/Wingless(non-core)</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>Q04771[21-123]</td>\n",
       "      <td>Q04771</td>\n",
       "      <td>Activin receptor type-1 (EC 2.7.11.30) (Activi...</td>\n",
       "      <td>ACVR1 ACVRLK2</td>\n",
       "      <td>509</td>\n",
       "      <td>glandular cells, smooth muscle cells</td>\n",
       "      <td>squamous epithelial cells, glial cells, pneumo...</td>\n",
       "      <td>myocytes, glandular cells, glandular cells, gl...</td>\n",
       "      <td>cells in endometrial stroma, glandular cells, ...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Q15797, Q99717</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>P36896[24-126]</td>\n",
       "      <td>P36896</td>\n",
       "      <td>Activin receptor type-1B (EC 2.7.11.30) (Activ...</td>\n",
       "      <td>ACVR1B ACVRLK4 ALK4</td>\n",
       "      <td>505</td>\n",
       "      <td>NaN</td>\n",
       "      <td>glandular cells, decidual cells, squamous epit...</td>\n",
       "      <td>exocrine glandular cells, glandular cells, gla...</td>\n",
       "      <td>melanocytes, Langerhans, keratinocytes, fibrob...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Q04771</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>2 rows  53 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       positionID   Entry                                      Protein names  \\\n",
       "0  Q04771[21-123]  Q04771  Activin receptor type-1 (EC 2.7.11.30) (Activi...   \n",
       "1  P36896[24-126]  P36896  Activin receptor type-1B (EC 2.7.11.30) (Activ...   \n",
       "\n",
       "            Gene names  Length             Protein Expression : High  \\\n",
       "0        ACVR1 ACVRLK2     509  glandular cells, smooth muscle cells   \n",
       "1  ACVR1B ACVRLK4 ALK4     505                                   NaN   \n",
       "\n",
       "                            Protein Expression : Low  \\\n",
       "0  squamous epithelial cells, glial cells, pneumo...   \n",
       "1  glandular cells, decidual cells, squamous epit...   \n",
       "\n",
       "                            Protein Expression : Med  \\\n",
       "0  myocytes, glandular cells, glandular cells, gl...   \n",
       "1  exocrine glandular cells, glandular cells, gla...   \n",
       "\n",
       "                                        Not detected Hedgehog(core)  ...  \\\n",
       "0  cells in endometrial stroma, glandular cells, ...            NaN  ...   \n",
       "1  melanocytes, Langerhans, keratinocytes, fibrob...            NaN  ...   \n",
       "\n",
       "  RTK(non-core),TGF(core) RTK(non-core),TGF(non-core)  \\\n",
       "0                     NaN                         NaN   \n",
       "1                     NaN                         NaN   \n",
       "\n",
       "  RTK(non-core),TGF(non-core),TGF(core),WNT/Wingless(core)  \\\n",
       "0                                                NaN         \n",
       "1                                                NaN         \n",
       "\n",
       "  RTK(non-core),WNT/Wingless(non-core)       TGF(core) TGF(non-core)  \\\n",
       "0                                  NaN  Q15797, Q99717           NaN   \n",
       "1                                  NaN          Q04771           NaN   \n",
       "\n",
       "  TGF(non-core),WNT/Wingless(core) TGF(non-core),WNT/Wingless(non-core)  \\\n",
       "0                              NaN                                  NaN   \n",
       "1                              NaN                                  NaN   \n",
       "\n",
       "  WNT/Wingless(core) WNT/Wingless(non-core)  \n",
       "0                NaN                    NaN  \n",
       "1                NaN                    NaN  \n",
       "\n",
       "[2 rows x 53 columns]"
      ]
     },
     "execution_count": 27,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "signal_data = SignaLink_1_0.loc[SignaLink_1_0['Entry'].isin(s2['Entry'])]\n",
    "signal_pathway=pd.merge(s2,SignaLink_1_0 ,how='inner', on=['Entry'])\n",
    "export_csv = signal_pathway.to_csv (r'/Users/saheeba/Downloads/SignaLink_sorted.csv', index = None, header=True)\n",
    "print(signal_pathway.shape)\n",
    "signal_pathway.head(2)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### <font color='blue'> xtalkdb: a database of signaling pathway crosstalk </font>\n",
    "xtalkdb contains curated information for 650 pairs of pathways from over 1600 publications. In addition, the database reports the molecular components (e.g. proteins, hormones, microRNAs) that mediate crosstalk between a pair of pathways and the species and tissue in which the crosstalk was observed"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(3474, 20)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Pathway A</th>\n",
       "      <th>Pathway B</th>\n",
       "      <th>Pubmed Query</th>\n",
       "      <th>PMID</th>\n",
       "      <th>Crosstalk</th>\n",
       "      <th>Transcriptional</th>\n",
       "      <th>Regulation type</th>\n",
       "      <th>Molecule A</th>\n",
       "      <th>Prot1</th>\n",
       "      <th>Molecule A Source</th>\n",
       "      <th>Molecule B</th>\n",
       "      <th>Prot2</th>\n",
       "      <th>Molecule B Source</th>\n",
       "      <th>Species</th>\n",
       "      <th>Tissue Name</th>\n",
       "      <th>BTO ID</th>\n",
       "      <th>Condition</th>\n",
       "      <th>Sentence from paper</th>\n",
       "      <th>Additional notes</th>\n",
       "      <th>Misleading sentences</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>Adipocytokine signaling pathway</td>\n",
       "      <td>ErbB signaling pathway</td>\n",
       "      <td>Adipocytokine ErbB</td>\n",
       "      <td>25838314</td>\n",
       "      <td>no</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Mentions both pathways but does not say taht t...</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>Adipocytokine signaling pathway</td>\n",
       "      <td>ErbB signaling pathway</td>\n",
       "      <td>Adipocytokine ErbB</td>\n",
       "      <td>23817051</td>\n",
       "      <td>no</td>\n",
       "      <td>no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>It is not mentioned that the pathways interact...</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                         Pathway A               Pathway B  \\\n",
       "0  Adipocytokine signaling pathway  ErbB signaling pathway   \n",
       "1  Adipocytokine signaling pathway  ErbB signaling pathway   \n",
       "\n",
       "         Pubmed Query      PMID Crosstalk Transcriptional Regulation type  \\\n",
       "0  Adipocytokine ErbB  25838314        no              no             NaN   \n",
       "1  Adipocytokine ErbB  23817051        no              no             NaN   \n",
       "\n",
       "  Molecule A Prot1 Molecule A Source Molecule B Prot2 Molecule B Source  \\\n",
       "0        NaN   NaN               NaN        NaN   NaN               NaN   \n",
       "1        NaN   NaN               NaN        NaN   NaN               NaN   \n",
       "\n",
       "  Species Tissue Name BTO ID Condition Sentence from paper  \\\n",
       "0     NaN         NaN    NaN       NaN                 NaN   \n",
       "1     NaN         NaN    NaN       NaN                 NaN   \n",
       "\n",
       "                                    Additional notes Misleading sentences  \n",
       "0  Mentions both pathways but does not say taht t...                  NaN  \n",
       "1  It is not mentioned that the pathways interact...                  NaN  "
      ]
     },
     "execution_count": 13,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "X_talk_db=pd.read_csv('X_talkdb.csv')\n",
    "print(X_talk_db.shape)\n",
    "X_talk_db.rename(columns={'Molecule A Identifier':'Prot1','Molecule B Identifier':'Prot2'},inplace=True)\n",
    "X_talk_db.head(2) "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "#To check if we miss anything in the X_talk database due to multiple entries in one row \n",
    "'''import re\n",
    "#re = re.split('; |, |\\/|:',str)\n",
    "list_prot1 = X_talk_db['Prot1'].str.split('; |, |\\/|:').tolist()\n",
    "s3.loc[s3['Entry'].isin(list_prot1)]\n",
    "true_false = pd.DataFrame(list_prot1).isin(s3['Entry'])\n",
    "true_false.to_csv('true_false_lits.txt')\n",
    "'''"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(268, 4)\n",
      "(312, 4)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th>Crosstalk</th>\n",
       "      <th>Entry</th>\n",
       "      <th>no</th>\n",
       "      <th>unclear</th>\n",
       "      <th>yes</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>A4K2Y1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Hippo signaling pathway</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>B0LXP3</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NF-kappa B signaling pathway</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2</td>\n",
       "      <td>B6V8E6</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Wnt signaling pathway</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>3</td>\n",
       "      <td>C4PCF3/A5HNF6</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Toll-like receptor signaling pathway</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>4</td>\n",
       "      <td>D3ZDU2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>mTOR signaling pathway</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "Crosstalk          Entry   no                               unclear  \\\n",
       "0                 A4K2Y1  NaN                                   NaN   \n",
       "1                 B0LXP3  NaN                                   NaN   \n",
       "2                 B6V8E6  NaN                                   NaN   \n",
       "3          C4PCF3/A5HNF6  NaN  Toll-like receptor signaling pathway   \n",
       "4                 D3ZDU2  NaN                                   NaN   \n",
       "\n",
       "Crosstalk                           yes  \n",
       "0               Hippo signaling pathway  \n",
       "1          NF-kappa B signaling pathway  \n",
       "2                 Wnt signaling pathway  \n",
       "3                                   NaN  \n",
       "4                mTOR signaling pathway  "
      ]
     },
     "execution_count": 14,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#Extract signaling pathways for each protein ID in X_talkdb\n",
    "\n",
    "X_talkprot1 = X_talk_db.groupby(['Prot1','Crosstalk'], as_index=False)['Pathway A'].agg(', '.join).set_index(['Prot1','Crosstalk'])['Pathway A'].unstack().reset_index()\n",
    "X_talkprot2 = X_talk_db.groupby(['Prot2','Crosstalk'], as_index=False)['Pathway B'].agg(', '.join).set_index(['Prot2','Crosstalk'])['Pathway B'].unstack().reset_index()\n",
    "X_talkprot1.rename(columns={'Prot1':'Entry'},inplace=True)\n",
    "X_talkprot2.rename(columns={'Prot2':'Entry'},inplace=True)\n",
    "print(X_talkprot1.shape)\n",
    "print(X_talkprot2.shape)\n",
    "X_talkprot2.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "X_talkprot3 = pd.merge(X_talkprot2,X_talkprot1, how='outer',on=None, left_on=None, right_on=None,\n",
    "         left_index=True, right_index=True, sort=True,\n",
    "         suffixes=('_x', '_y'), copy=True, indicator=True,\n",
    "         validate=\"1:1\")\n",
    "print(X_talkprot3.shape)\n",
    "X_talkprot3.head(20)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "X_talkprot3 = X_talk_db.groupby(['Prot1','Prot2'], as_index=False)['Pathway A'].agg(', '.join).set_index(['Prot1','Prot2'])['Pathway A'].unstack().reset_index(drop=True)\n",
    "\n",
    "'''\n",
    "X_talkprot3 = pd.merge(X_talkprot1,X_talkprot2, how='inner',on=None, left_on=None, right_on=None,\n",
    "         left_index=True, right_index=True, sort=True,\n",
    "         suffixes=('_x', '_y'), copy=True, indicator=True,\n",
    "         validate=\"1:1\")\n",
    "print(X_talkprot3.shape)\n",
    "X_talkprot3.head()\n",
    "'''\n",
    "X_talkprot3.head(2)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "print(X_talkprot1.shape)\n",
    "print(X_talkprot2.shape)\n",
    "X_talkprot1.head()\n",
    "crosstalk = pd.concat([X_talkprot1,X_talkprot2])\n",
    "print(crosstalk.shape)\n",
    "crosstalk.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "crosstalk = crosstalk.loc[crosstalk['Entry'].isin(s2['Entry'])]\n",
    "print(crosstalk.shape)\n",
    "crosstalk.head().reset_index(drop=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "#another way of doing same match again\n",
    "prot1_pathway = X_talkprot1.loc[X_talkprot1['Entry'].isin(s2['Entry'])]\n",
    "prot2_pathway = X_talkprot2.loc[X_talkprot2['Entry'].isin(s2['Entry'])]\n",
    "prot1_prot2 = pd.concat([prot1_pathway,prot2_pathway])\n",
    "print(prot1_prot2.shape)\n",
    "#prot1_prot2['Entry'] = prot1_prot2['Entry'].str.split(\" \").str[0]#.str.split(\" \") \n",
    "prot1_prot2.head()#['Crosstalk']#.drop(['Crosstalk'], axis = 1)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## <font color='green'> Extracting signaling pathway from Uniprot dataset </font>\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "!egrep -w 'signaling|pathway' 997_from\\ uniprot.gff > signaling_pathway_uniprot.csv\n",
    "!egrep -w 'signaling' 997_from\\ uniprot.gff > signaling_uniprot.csv\n",
    "!egrep -w 'pathway' 997_from\\ uniprot.gff > pathway_uniprot.csv"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "signaling_pathway_uniprot = pd.read_csv('signaling_pathway_uniprot.csv',sep=\"\\t\",header= None)\n",
    "print(signaling_pathway_uniprot.shape)\n",
    "signaling_uniprot = pd.read_csv('signaling_uniprot.csv',sep=\"\\t\",header= None)\n",
    "print(signaling_uniprot.shape)\n",
    "pathway_uniprot = pd.read_csv('pathway_uniprot.csv',sep=\"\\t\",header= None)\n",
    "print(pathway_uniprot.shape)\n",
    "signaling_pathway_uniprot"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "import re\n",
    "found = []\n",
    "match = re.compile('^(\\S+).*?''(pathway)')\n",
    "with open(\"997_from uniprot.gff\", \"r\") as f:\n",
    "    searchlines = f.readlines()\n",
    "j=len(searchlines)-1 #with open('997_from uniprot.gff', \"r\") as raw:\n",
    "for i, line in enumerate(searchlines):  #  for rec in raw:\n",
    "        found.append(line)#line)#\n",
    "#print(pd.DataFrame(found).shape)\n",
    "#pd.DataFrame(found).head()\n",
    "pd.DataFrame(found)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "'''with open(\"997_from uniprot.gff\", \"r\") as f:\n",
    "    searchlines = f.readlines()\n",
    "    for i, line in enumerate(searchlines):\n",
    "        print(line)'''\n",
    "found = []\n",
    "match = re.compile('^(\\S+).*?(\\w*)'\" \"'(signal)')\n",
    "with open(\"997_from uniprot.gff\", \"r\") as f:\n",
    "    searchlines = f.readlines()\n",
    "j=len(searchlines)-1 #with open('997_from uniprot.gff', \"r\") as raw:\n",
    "for i, line in enumerate(searchlines):  #  for rec in raw:\n",
    "    found.extend(match.findall(line))\n",
    "print(pd.DataFrame(found).shape)\n",
    "pd.DataFrame(found).head() "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "columns_names = ['Entry','one','two','three','f','fi','si','se','word_search','none']\n",
    "uni_signal = pd.read_csv('signaling_pathway_uniprot.csv',sep=\"\\t\",header= None)\n",
    "uni_signal.rename(columns={int(0):'Entry',int(8):'word_search'}, inplace=True)\n",
    "uni_signal.drop(uni_signal.columns[[1,2,3,4,5,6,7,9]], axis=1, inplace=True)\n",
    "print(uni_signal.shape)\n",
    "uni_signal.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### <font color='green'> Extracting signaling pathway from Reactome dataset </font>\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(297048, 6)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Entry</th>\n",
       "      <th>1</th>\n",
       "      <th>2</th>\n",
       "      <th>Reactome</th>\n",
       "      <th>4</th>\n",
       "      <th>5</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>A0A023GRW3</td>\n",
       "      <td>R-DME-72163</td>\n",
       "      <td>https://reactome.org/PathwayBrowser/#/R-DME-72163</td>\n",
       "      <td>mRNA Splicing - Major Pathway</td>\n",
       "      <td>IEA</td>\n",
       "      <td>Drosophila melanogaster</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>A0A023IKK2</td>\n",
       "      <td>R-RNO-202424</td>\n",
       "      <td>https://reactome.org/PathwayBrowser/#/R-RNO-20...</td>\n",
       "      <td>Downstream TCR signaling</td>\n",
       "      <td>IEA</td>\n",
       "      <td>Rattus norvegicus</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2</td>\n",
       "      <td>A0A023IKK2</td>\n",
       "      <td>R-RNO-202427</td>\n",
       "      <td>https://reactome.org/PathwayBrowser/#/R-RNO-20...</td>\n",
       "      <td>Phosphorylation of CD3 and TCR zeta chains</td>\n",
       "      <td>IEA</td>\n",
       "      <td>Rattus norvegicus</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "        Entry             1  \\\n",
       "0  A0A023GRW3   R-DME-72163   \n",
       "1  A0A023IKK2  R-RNO-202424   \n",
       "2  A0A023IKK2  R-RNO-202427   \n",
       "\n",
       "                                                   2  \\\n",
       "0  https://reactome.org/PathwayBrowser/#/R-DME-72163   \n",
       "1  https://reactome.org/PathwayBrowser/#/R-RNO-20...   \n",
       "2  https://reactome.org/PathwayBrowser/#/R-RNO-20...   \n",
       "\n",
       "                                     Reactome    4                        5  \n",
       "0               mRNA Splicing - Major Pathway  IEA  Drosophila melanogaster  \n",
       "1                    Downstream TCR signaling  IEA        Rattus norvegicus  \n",
       "2  Phosphorylation of CD3 and TCR zeta chains  IEA        Rattus norvegicus  "
      ]
     },
     "execution_count": 15,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "reactome = pd.read_csv('UniProt2Reactome.txt',sep=\"\\t\",header= None)\n",
    "print(reactome.shape)\n",
    "reactome.rename(columns={int(0):'Entry',int(3):'Reactome'}, inplace=True)\n",
    "#pd.merge(reactome,s3 ,how='inner', on=['Entry'])\n",
    "reactome.head(3)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 137,
   "metadata": {},
   "outputs": [],
   "source": [
    "match_reactome = reactome.loc[reactome['Entry'].isin(s3['Entry'])]\n",
    "pos_match = match_reactome.groupby(['Entry'], as_index=False)['Reactome'].agg(\",\".join)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Entry</th>\n",
       "      <th>Reactome</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>O00206</td>\n",
       "      <td>ER-Phagosome pathway,Caspase activation via De...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>O00220</td>\n",
       "      <td>Caspase activation via Death Receptors in the ...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2</td>\n",
       "      <td>O00241</td>\n",
       "      <td>DAP12 interactions,Signal regulatory protein f...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>3</td>\n",
       "      <td>O00478</td>\n",
       "      <td>Butyrophilin (BTN) family interactions</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>4</td>\n",
       "      <td>O00481</td>\n",
       "      <td>Butyrophilin (BTN) family interactions</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "    Entry                                           Reactome\n",
       "0  O00206  ER-Phagosome pathway,Caspase activation via De...\n",
       "1  O00220  Caspase activation via Death Receptors in the ...\n",
       "2  O00241  DAP12 interactions,Signal regulatory protein f...\n",
       "3  O00478             Butyrophilin (BTN) family interactions\n",
       "4  O00481             Butyrophilin (BTN) family interactions"
      ]
     },
     "execution_count": 17,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "pos_match.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "#uni_signal[\"word_search\"].str.split(\"signaling pathway\", n = 1, expand = True)     #data[\"Team\"].str.rsplit(\"t\", n = 1, expand = False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### merging files for complete dataset"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 93,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(438, 11)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>positionID</th>\n",
       "      <th>Entry</th>\n",
       "      <th>sequences</th>\n",
       "      <th>Protein names</th>\n",
       "      <th>Gene names</th>\n",
       "      <th>Seq_length</th>\n",
       "      <th>Protein Expression : High</th>\n",
       "      <th>Protein Expression : Low</th>\n",
       "      <th>Protein Expression : Med</th>\n",
       "      <th>Not detected</th>\n",
       "      <th>Reactome</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>P12821[30-1256]</td>\n",
       "      <td>P12821</td>\n",
       "      <td>LDPGLQPGNFSADEAGAQLFAQSYNSSAEQVLFQSVAASWAHDTNI...</td>\n",
       "      <td>Angiotensin-converting enzyme (ACE) (EC 3.2.1....</td>\n",
       "      <td>ACE DCP DCP1</td>\n",
       "      <td>1227</td>\n",
       "      <td>glandular cells, glandular cells, glandular ce...</td>\n",
       "      <td>endothelial cells</td>\n",
       "      <td>glandular cells, pneumocytes, cells in seminif...</td>\n",
       "      <td>fibroblasts, glandular cells, peripheral nerve...</td>\n",
       "      <td>Metabolism of Angiotensinogen to Angiotensins</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>Q9BYF1[18-740]</td>\n",
       "      <td>Q9BYF1</td>\n",
       "      <td>QSTIEEQAKTFLDKFNHEAEDLFYQSSLASWNYNTNITEENVQNMN...</td>\n",
       "      <td>Angiotensin-converting enzyme 2 (EC 3.4.17.23)...</td>\n",
       "      <td>ACE2 UNQ868/PRO1885</td>\n",
       "      <td>723</td>\n",
       "      <td>glandular cells, Leydig cells, cells in semini...</td>\n",
       "      <td>glandular cells, glandular cells, glandular ce...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>adipocytes, smooth muscle cells, epidermal cel...</td>\n",
       "      <td>Metabolism of Angiotensinogen to Angiotensins</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "        positionID   Entry                                          sequences  \\\n",
       "0  P12821[30-1256]  P12821  LDPGLQPGNFSADEAGAQLFAQSYNSSAEQVLFQSVAASWAHDTNI...   \n",
       "1   Q9BYF1[18-740]  Q9BYF1  QSTIEEQAKTFLDKFNHEAEDLFYQSSLASWNYNTNITEENVQNMN...   \n",
       "\n",
       "                                       Protein names           Gene names  \\\n",
       "0  Angiotensin-converting enzyme (ACE) (EC 3.2.1....         ACE DCP DCP1   \n",
       "1  Angiotensin-converting enzyme 2 (EC 3.4.17.23)...  ACE2 UNQ868/PRO1885   \n",
       "\n",
       "   Seq_length                          Protein Expression : High  \\\n",
       "0        1227  glandular cells, glandular cells, glandular ce...   \n",
       "1         723  glandular cells, Leydig cells, cells in semini...   \n",
       "\n",
       "                            Protein Expression : Low  \\\n",
       "0                                  endothelial cells   \n",
       "1  glandular cells, glandular cells, glandular ce...   \n",
       "\n",
       "                            Protein Expression : Med  \\\n",
       "0  glandular cells, pneumocytes, cells in seminif...   \n",
       "1                                                NaN   \n",
       "\n",
       "                                        Not detected  \\\n",
       "0  fibroblasts, glandular cells, peripheral nerve...   \n",
       "1  adipocytes, smooth muscle cells, epidermal cel...   \n",
       "\n",
       "                                        Reactome  \n",
       "0  Metabolism of Angiotensinogen to Angiotensins  \n",
       "1  Metabolism of Angiotensinogen to Angiotensins  "
      ]
     },
     "execution_count": 93,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "Reactome_signal_pathway=pd.merge(s3,pos_match ,how='inner', on=['Entry'])\n",
    "print(Reactome_signal_pathway.shape)\n",
    "Reactome_signal_pathway.to_csv(r'Dataset_with_reactome.csv', index = None, header=True)#(\n",
    "Reactome_signal_pathway.head(2)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 88,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(21, 11)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>positionID</th>\n",
       "      <th>Entry</th>\n",
       "      <th>sequences</th>\n",
       "      <th>Protein names</th>\n",
       "      <th>Gene names</th>\n",
       "      <th>Seq_length</th>\n",
       "      <th>Protein Expression : High</th>\n",
       "      <th>Protein Expression : Low</th>\n",
       "      <th>Protein Expression : Med</th>\n",
       "      <th>Not detected</th>\n",
       "      <th>X_talk pathway</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>O14672[20-672]</td>\n",
       "      <td>O14672</td>\n",
       "      <td>QYGNPLNKYIRHYEGLSYNVDSLHQKHQRAKRAVSHEDQFLRLDFH...</td>\n",
       "      <td>Disintegrin and metalloproteinase domain-conta...</td>\n",
       "      <td>ADAM10 KUZ MADM</td>\n",
       "      <td>653</td>\n",
       "      <td>NaN</td>\n",
       "      <td>glandular cells, decidual cells, squamous epit...</td>\n",
       "      <td>fibroblasts, glandular cells, glandular cells,...</td>\n",
       "      <td>myocytes, glandular cells, trophoblastic cells...</td>\n",
       "      <td>Notch signaling pathway, Notch signaling pathway</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>Q13145[21-152]</td>\n",
       "      <td>Q13145</td>\n",
       "      <td>VLLTKGEIRCYCDAAHCVATGYMCKSELSACFSRLLDPQNSNSPLT...</td>\n",
       "      <td>BMP and activin membrane-bound inhibitor homol...</td>\n",
       "      <td>BAMBI NMA</td>\n",
       "      <td>132</td>\n",
       "      <td>NaN</td>\n",
       "      <td>glandular cells, glandular cells, decidual cel...</td>\n",
       "      <td>glandular cells, respiratory epithelial cells,...</td>\n",
       "      <td>glandular cells, trophoblastic cells, ovarian ...</td>\n",
       "      <td>TGF-beta signaling pathway</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "       positionID   Entry                                          sequences  \\\n",
       "0  O14672[20-672]  O14672  QYGNPLNKYIRHYEGLSYNVDSLHQKHQRAKRAVSHEDQFLRLDFH...   \n",
       "1  Q13145[21-152]  Q13145  VLLTKGEIRCYCDAAHCVATGYMCKSELSACFSRLLDPQNSNSPLT...   \n",
       "\n",
       "                                       Protein names       Gene names  \\\n",
       "0  Disintegrin and metalloproteinase domain-conta...  ADAM10 KUZ MADM   \n",
       "1  BMP and activin membrane-bound inhibitor homol...        BAMBI NMA   \n",
       "\n",
       "   Seq_length Protein Expression : High  \\\n",
       "0         653                       NaN   \n",
       "1         132                       NaN   \n",
       "\n",
       "                            Protein Expression : Low  \\\n",
       "0  glandular cells, decidual cells, squamous epit...   \n",
       "1  glandular cells, glandular cells, decidual cel...   \n",
       "\n",
       "                            Protein Expression : Med  \\\n",
       "0  fibroblasts, glandular cells, glandular cells,...   \n",
       "1  glandular cells, respiratory epithelial cells,...   \n",
       "\n",
       "                                        Not detected  \\\n",
       "0  myocytes, glandular cells, trophoblastic cells...   \n",
       "1  glandular cells, trophoblastic cells, ovarian ...   \n",
       "\n",
       "                                     X_talk pathway  \n",
       "0  Notch signaling pathway, Notch signaling pathway  \n",
       "1                        TGF-beta signaling pathway  "
      ]
     },
     "execution_count": 88,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "X_talk_db_sorted = X_talkprot2[['Entry', 'yes']]\n",
    "X_talk_db_sorted.rename(columns={'yes':'X_talk pathway'},inplace=True)\n",
    "X_talk_signal_pathway=pd.merge(s3, X_talk_db_sorted ,how='inner', on=['Entry'])\n",
    "X_talk_signal_pathway.to_csv(r'Dataset_with_X_talk.csv', index = None, header=True)#(r'/Users/saheeba/Downloads/notlisted_extracellular.csv',)\n",
    "print(X_talk_signal_pathway.shape)\n",
    "X_talk_signal_pathway.head(2)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 73,
   "metadata": {},
   "outputs": [],
   "source": [
    "#['bar'].fillna(df['foo'], inplace=True)\n",
    "#del df['foo']\n",
    "SignaLink_source=pd.DataFrame(source_match.rename(columns={'Source':'Entry','source_pathways':'SignaLink Pathway'}, inplace=True))\n",
    "SignaLink_target = pd.DataFrame(target_match.rename(columns={'Target':'Entry','target_pathways':'SignaLink Pathway'}, inplace=True))\n",
    "#print(SignaLink_source.iloc[:,1].tolist())#\n",
    "SignaLink_source.append(SignaLink_target)\n",
    "SignaLink_source.rename(columns={ 0:'SignaLink'},inplace=True)\n",
    "#SignaLink_merge_pathway=pd.merge(s3, SignaLink_source ,how='inner', on=['Entry'])\n",
    "#print(SignaLink_merge_pathway.shape)\n",
    "#SignaLink_merge_pathway.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 94,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(60, 11)\n"
     ]
    }
   ],
   "source": [
    "SignaLink_source.to_csv(r'SignaLink_source.csv')\n",
    "SignaLink_match = pd.read_csv(\"SignaLink_source.csv\")\n",
    "SignaLink_match.rename(columns={'Source':'Entry'},inplace=True)\n",
    "SignaLink_merge_pathway=pd.merge(s3, SignaLink_match ,how='inner', on=['Entry'])\n",
    "SignaLink_merge_pathway = SignaLink_merge_pathway.drop(columns=['source_pathways'])\n",
    "SignaLink_merge_pathway.to_csv(r'Dataset_with_SignaLink.csv', index = None, header=True)\n",
    "print(SignaLink_merge_pathway.shape)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 129,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(519, 13)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Entry</th>\n",
       "      <th>Gene names</th>\n",
       "      <th>Not detected</th>\n",
       "      <th>Protein Expression : High</th>\n",
       "      <th>Protein Expression : Low</th>\n",
       "      <th>Protein Expression : Med</th>\n",
       "      <th>Protein names</th>\n",
       "      <th>Reactome</th>\n",
       "      <th>Seq_length</th>\n",
       "      <th>SignaLink</th>\n",
       "      <th>X_talk pathway</th>\n",
       "      <th>positionID</th>\n",
       "      <th>sequences</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>Q04771</td>\n",
       "      <td>ACVR1 ACVRLK2</td>\n",
       "      <td>cells in endometrial stroma, glandular cells, ...</td>\n",
       "      <td>glandular cells, smooth muscle cells</td>\n",
       "      <td>squamous epithelial cells, glial cells, pneumo...</td>\n",
       "      <td>myocytes, glandular cells, glandular cells, gl...</td>\n",
       "      <td>Activin receptor type-1 (EC 2.7.11.30) (Activi...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>103</td>\n",
       "      <td>TGF(core),TGF(core)</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Q04771[21-123]</td>\n",
       "      <td>MEDEKPKVNPKLYMCVCEGLSCGNEDHCEGQQCFSSLSINDGFHVY...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "    Entry     Gene names                                       Not detected  \\\n",
       "0  Q04771  ACVR1 ACVRLK2  cells in endometrial stroma, glandular cells, ...   \n",
       "\n",
       "              Protein Expression : High  \\\n",
       "0  glandular cells, smooth muscle cells   \n",
       "\n",
       "                            Protein Expression : Low  \\\n",
       "0  squamous epithelial cells, glial cells, pneumo...   \n",
       "\n",
       "                            Protein Expression : Med  \\\n",
       "0  myocytes, glandular cells, glandular cells, gl...   \n",
       "\n",
       "                                       Protein names Reactome  Seq_length  \\\n",
       "0  Activin receptor type-1 (EC 2.7.11.30) (Activi...      NaN         103   \n",
       "\n",
       "             SignaLink X_talk pathway      positionID  \\\n",
       "0  TGF(core),TGF(core)            NaN  Q04771[21-123]   \n",
       "\n",
       "                                           sequences  \n",
       "0  MEDEKPKVNPKLYMCVCEGLSCGNEDHCEGQQCFSSLSINDGFHVY...  "
      ]
     },
     "execution_count": 129,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#pd.merge(SignaLink_merge_pathway,X_talk_db_sorted ,how='outer', on=['Entry'])\n",
    "together = pd.concat([SignaLink_merge_pathway, X_talk_signal_pathway,Reactome_signal_pathway], sort=True)#.shape\n",
    "together.to_csv(r\"together_all_pathway.csv\")\n",
    "print(together.shape)\n",
    "together.head(1)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 130,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(519, 12)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Entry</th>\n",
       "      <th>Gene names</th>\n",
       "      <th>Not detected</th>\n",
       "      <th>Protein Expression : High</th>\n",
       "      <th>Protein Expression : Low</th>\n",
       "      <th>Protein Expression : Med</th>\n",
       "      <th>Protein names</th>\n",
       "      <th>Reactome</th>\n",
       "      <th>Seq_length</th>\n",
       "      <th>X_talk pathway</th>\n",
       "      <th>positionID</th>\n",
       "      <th>sequences</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>Q04771</td>\n",
       "      <td>ACVR1 ACVRLK2</td>\n",
       "      <td>cells in endometrial stroma, glandular cells, ...</td>\n",
       "      <td>glandular cells, smooth muscle cells</td>\n",
       "      <td>squamous epithelial cells, glial cells, pneumo...</td>\n",
       "      <td>myocytes, glandular cells, glandular cells, gl...</td>\n",
       "      <td>Activin receptor type-1 (EC 2.7.11.30) (Activi...</td>\n",
       "      <td>TGF(core),TGF(core)</td>\n",
       "      <td>103</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Q04771[21-123]</td>\n",
       "      <td>MEDEKPKVNPKLYMCVCEGLSCGNEDHCEGQQCFSSLSINDGFHVY...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>P36896</td>\n",
       "      <td>ACVR1B ACVRLK4 ALK4</td>\n",
       "      <td>melanocytes, Langerhans, keratinocytes, fibrob...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>glandular cells, decidual cells, squamous epit...</td>\n",
       "      <td>exocrine glandular cells, glandular cells, gla...</td>\n",
       "      <td>Activin receptor type-1B (EC 2.7.11.30) (Activ...</td>\n",
       "      <td>TGF(core),TGF(core)</td>\n",
       "      <td>103</td>\n",
       "      <td>NaN</td>\n",
       "      <td>P36896[24-126]</td>\n",
       "      <td>SGPRGVQALLCACTSCLQANYTCETDGACMVSIFNLDGMEHHVRTC...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2</td>\n",
       "      <td>Q8NER5</td>\n",
       "      <td>ACVR1C ALK7</td>\n",
       "      <td>Langerhans</td>\n",
       "      <td>glandular cells, cells in tubules, glandular c...</td>\n",
       "      <td>glial cells, hematopoietic cells, adipocytes, ...</td>\n",
       "      <td>myoepithelial cells, cells in molecular layer,...</td>\n",
       "      <td>Activin receptor type-1C (EC 2.7.11.30) (Activ...</td>\n",
       "      <td>TGF(core)</td>\n",
       "      <td>92</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Q8NER5[22-113]</td>\n",
       "      <td>LSPGLKCVCLLCDSSNFTCQTEGACWASVMLTNGKEQVIKSCVSLP...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>3</td>\n",
       "      <td>Q13705</td>\n",
       "      <td>ACVR2B</td>\n",
       "      <td>cells in endometrial stroma, cells in endometr...</td>\n",
       "      <td>chondrocytes</td>\n",
       "      <td>endothelial cells, glandular cells, peripheral...</td>\n",
       "      <td>glandular cells, respiratory epithelial cells,...</td>\n",
       "      <td>Activin receptor type-2B (EC 2.7.11.30) (Activ...</td>\n",
       "      <td>TGF(core),TGF(core),TGF(core),TGF(core),TGF(core)</td>\n",
       "      <td>119</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Q13705[19-137]</td>\n",
       "      <td>SGRGEAETRECIYYNANWELERTNQSGLERCEGEQDKRLHCYASWR...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>4</td>\n",
       "      <td>P37023</td>\n",
       "      <td>ACVRL1 ACVRLK1 ALK1</td>\n",
       "      <td>squamous epithelial cells, peripheral nerve/ga...</td>\n",
       "      <td>glandular cells, neuronal cells, cells in tubu...</td>\n",
       "      <td>glandular cells, glial cells, glandular cells,...</td>\n",
       "      <td>macrophages, cells in molecular layer, Purkinj...</td>\n",
       "      <td>Serine/threonine-protein kinase receptor R3 (S...</td>\n",
       "      <td>TGF(core),TGF(core),TGF(core)</td>\n",
       "      <td>97</td>\n",
       "      <td>NaN</td>\n",
       "      <td>P37023[22-118]</td>\n",
       "      <td>DPVKPSRGPLVTCTCESPHCKGPTCRGAWCTVVLVREEGRHPQEHR...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "    Entry           Gene names  \\\n",
       "0  Q04771        ACVR1 ACVRLK2   \n",
       "1  P36896  ACVR1B ACVRLK4 ALK4   \n",
       "2  Q8NER5          ACVR1C ALK7   \n",
       "3  Q13705               ACVR2B   \n",
       "4  P37023  ACVRL1 ACVRLK1 ALK1   \n",
       "\n",
       "                                        Not detected  \\\n",
       "0  cells in endometrial stroma, glandular cells, ...   \n",
       "1  melanocytes, Langerhans, keratinocytes, fibrob...   \n",
       "2                                         Langerhans   \n",
       "3  cells in endometrial stroma, cells in endometr...   \n",
       "4  squamous epithelial cells, peripheral nerve/ga...   \n",
       "\n",
       "                           Protein Expression : High  \\\n",
       "0               glandular cells, smooth muscle cells   \n",
       "1                                                NaN   \n",
       "2  glandular cells, cells in tubules, glandular c...   \n",
       "3                                       chondrocytes   \n",
       "4  glandular cells, neuronal cells, cells in tubu...   \n",
       "\n",
       "                            Protein Expression : Low  \\\n",
       "0  squamous epithelial cells, glial cells, pneumo...   \n",
       "1  glandular cells, decidual cells, squamous epit...   \n",
       "2  glial cells, hematopoietic cells, adipocytes, ...   \n",
       "3  endothelial cells, glandular cells, peripheral...   \n",
       "4  glandular cells, glial cells, glandular cells,...   \n",
       "\n",
       "                            Protein Expression : Med  \\\n",
       "0  myocytes, glandular cells, glandular cells, gl...   \n",
       "1  exocrine glandular cells, glandular cells, gla...   \n",
       "2  myoepithelial cells, cells in molecular layer,...   \n",
       "3  glandular cells, respiratory epithelial cells,...   \n",
       "4  macrophages, cells in molecular layer, Purkinj...   \n",
       "\n",
       "                                       Protein names  \\\n",
       "0  Activin receptor type-1 (EC 2.7.11.30) (Activi...   \n",
       "1  Activin receptor type-1B (EC 2.7.11.30) (Activ...   \n",
       "2  Activin receptor type-1C (EC 2.7.11.30) (Activ...   \n",
       "3  Activin receptor type-2B (EC 2.7.11.30) (Activ...   \n",
       "4  Serine/threonine-protein kinase receptor R3 (S...   \n",
       "\n",
       "                                            Reactome  Seq_length  \\\n",
       "0                                TGF(core),TGF(core)         103   \n",
       "1                                TGF(core),TGF(core)         103   \n",
       "2                                          TGF(core)          92   \n",
       "3  TGF(core),TGF(core),TGF(core),TGF(core),TGF(core)         119   \n",
       "4                      TGF(core),TGF(core),TGF(core)          97   \n",
       "\n",
       "  X_talk pathway      positionID  \\\n",
       "0            NaN  Q04771[21-123]   \n",
       "1            NaN  P36896[24-126]   \n",
       "2            NaN  Q8NER5[22-113]   \n",
       "3            NaN  Q13705[19-137]   \n",
       "4            NaN  P37023[22-118]   \n",
       "\n",
       "                                           sequences  \n",
       "0  MEDEKPKVNPKLYMCVCEGLSCGNEDHCEGQQCFSSLSINDGFHVY...  \n",
       "1  SGPRGVQALLCACTSCLQANYTCETDGACMVSIFNLDGMEHHVRTC...  \n",
       "2  LSPGLKCVCLLCDSSNFTCQTEGACWASVMLTNGKEQVIKSCVSLP...  \n",
       "3  SGRGEAETRECIYYNANWELERTNQSGLERCEGEQDKRLHCYASWR...  \n",
       "4  DPVKPSRGPLVTCTCESPHCKGPTCRGAWCTVVLVREEGRHPQEHR...  "
      ]
     },
     "execution_count": 130,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "together['Reactome'].fillna(together['SignaLink'], inplace=True)\n",
    "del together['SignaLink']\n",
    "print(together.shape)\n",
    "together.head(5)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 133,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(519, 11)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Entry</th>\n",
       "      <th>Gene names</th>\n",
       "      <th>Not detected</th>\n",
       "      <th>Protein Expression : High</th>\n",
       "      <th>Protein Expression : Low</th>\n",
       "      <th>Protein Expression : Med</th>\n",
       "      <th>Protein names</th>\n",
       "      <th>Reactome</th>\n",
       "      <th>Seq_length</th>\n",
       "      <th>positionID</th>\n",
       "      <th>sequences</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>Q04771</td>\n",
       "      <td>ACVR1 ACVRLK2</td>\n",
       "      <td>cells in endometrial stroma, glandular cells, ...</td>\n",
       "      <td>glandular cells, smooth muscle cells</td>\n",
       "      <td>squamous epithelial cells, glial cells, pneumo...</td>\n",
       "      <td>myocytes, glandular cells, glandular cells, gl...</td>\n",
       "      <td>Activin receptor type-1 (EC 2.7.11.30) (Activi...</td>\n",
       "      <td>TGF(core),TGF(core)</td>\n",
       "      <td>103</td>\n",
       "      <td>Q04771[21-123]</td>\n",
       "      <td>MEDEKPKVNPKLYMCVCEGLSCGNEDHCEGQQCFSSLSINDGFHVY...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>P36896</td>\n",
       "      <td>ACVR1B ACVRLK4 ALK4</td>\n",
       "      <td>melanocytes, Langerhans, keratinocytes, fibrob...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>glandular cells, decidual cells, squamous epit...</td>\n",
       "      <td>exocrine glandular cells, glandular cells, gla...</td>\n",
       "      <td>Activin receptor type-1B (EC 2.7.11.30) (Activ...</td>\n",
       "      <td>TGF(core),TGF(core)</td>\n",
       "      <td>103</td>\n",
       "      <td>P36896[24-126]</td>\n",
       "      <td>SGPRGVQALLCACTSCLQANYTCETDGACMVSIFNLDGMEHHVRTC...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2</td>\n",
       "      <td>Q8NER5</td>\n",
       "      <td>ACVR1C ALK7</td>\n",
       "      <td>Langerhans</td>\n",
       "      <td>glandular cells, cells in tubules, glandular c...</td>\n",
       "      <td>glial cells, hematopoietic cells, adipocytes, ...</td>\n",
       "      <td>myoepithelial cells, cells in molecular layer,...</td>\n",
       "      <td>Activin receptor type-1C (EC 2.7.11.30) (Activ...</td>\n",
       "      <td>TGF(core)</td>\n",
       "      <td>92</td>\n",
       "      <td>Q8NER5[22-113]</td>\n",
       "      <td>LSPGLKCVCLLCDSSNFTCQTEGACWASVMLTNGKEQVIKSCVSLP...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>3</td>\n",
       "      <td>Q13705</td>\n",
       "      <td>ACVR2B</td>\n",
       "      <td>cells in endometrial stroma, cells in endometr...</td>\n",
       "      <td>chondrocytes</td>\n",
       "      <td>endothelial cells, glandular cells, peripheral...</td>\n",
       "      <td>glandular cells, respiratory epithelial cells,...</td>\n",
       "      <td>Activin receptor type-2B (EC 2.7.11.30) (Activ...</td>\n",
       "      <td>TGF(core),TGF(core),TGF(core),TGF(core),TGF(core)</td>\n",
       "      <td>119</td>\n",
       "      <td>Q13705[19-137]</td>\n",
       "      <td>SGRGEAETRECIYYNANWELERTNQSGLERCEGEQDKRLHCYASWR...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>4</td>\n",
       "      <td>P37023</td>\n",
       "      <td>ACVRL1 ACVRLK1 ALK1</td>\n",
       "      <td>squamous epithelial cells, peripheral nerve/ga...</td>\n",
       "      <td>glandular cells, neuronal cells, cells in tubu...</td>\n",
       "      <td>glandular cells, glial cells, glandular cells,...</td>\n",
       "      <td>macrophages, cells in molecular layer, Purkinj...</td>\n",
       "      <td>Serine/threonine-protein kinase receptor R3 (S...</td>\n",
       "      <td>TGF(core),TGF(core),TGF(core)</td>\n",
       "      <td>97</td>\n",
       "      <td>P37023[22-118]</td>\n",
       "      <td>DPVKPSRGPLVTCTCESPHCKGPTCRGAWCTVVLVREEGRHPQEHR...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "    Entry           Gene names  \\\n",
       "0  Q04771        ACVR1 ACVRLK2   \n",
       "1  P36896  ACVR1B ACVRLK4 ALK4   \n",
       "2  Q8NER5          ACVR1C ALK7   \n",
       "3  Q13705               ACVR2B   \n",
       "4  P37023  ACVRL1 ACVRLK1 ALK1   \n",
       "\n",
       "                                        Not detected  \\\n",
       "0  cells in endometrial stroma, glandular cells, ...   \n",
       "1  melanocytes, Langerhans, keratinocytes, fibrob...   \n",
       "2                                         Langerhans   \n",
       "3  cells in endometrial stroma, cells in endometr...   \n",
       "4  squamous epithelial cells, peripheral nerve/ga...   \n",
       "\n",
       "                           Protein Expression : High  \\\n",
       "0               glandular cells, smooth muscle cells   \n",
       "1                                                NaN   \n",
       "2  glandular cells, cells in tubules, glandular c...   \n",
       "3                                       chondrocytes   \n",
       "4  glandular cells, neuronal cells, cells in tubu...   \n",
       "\n",
       "                            Protein Expression : Low  \\\n",
       "0  squamous epithelial cells, glial cells, pneumo...   \n",
       "1  glandular cells, decidual cells, squamous epit...   \n",
       "2  glial cells, hematopoietic cells, adipocytes, ...   \n",
       "3  endothelial cells, glandular cells, peripheral...   \n",
       "4  glandular cells, glial cells, glandular cells,...   \n",
       "\n",
       "                            Protein Expression : Med  \\\n",
       "0  myocytes, glandular cells, glandular cells, gl...   \n",
       "1  exocrine glandular cells, glandular cells, gla...   \n",
       "2  myoepithelial cells, cells in molecular layer,...   \n",
       "3  glandular cells, respiratory epithelial cells,...   \n",
       "4  macrophages, cells in molecular layer, Purkinj...   \n",
       "\n",
       "                                       Protein names  \\\n",
       "0  Activin receptor type-1 (EC 2.7.11.30) (Activi...   \n",
       "1  Activin receptor type-1B (EC 2.7.11.30) (Activ...   \n",
       "2  Activin receptor type-1C (EC 2.7.11.30) (Activ...   \n",
       "3  Activin receptor type-2B (EC 2.7.11.30) (Activ...   \n",
       "4  Serine/threonine-protein kinase receptor R3 (S...   \n",
       "\n",
       "                                            Reactome  Seq_length  \\\n",
       "0                                TGF(core),TGF(core)         103   \n",
       "1                                TGF(core),TGF(core)         103   \n",
       "2                                          TGF(core)          92   \n",
       "3  TGF(core),TGF(core),TGF(core),TGF(core),TGF(core)         119   \n",
       "4                      TGF(core),TGF(core),TGF(core)          97   \n",
       "\n",
       "       positionID                                          sequences  \n",
       "0  Q04771[21-123]  MEDEKPKVNPKLYMCVCEGLSCGNEDHCEGQQCFSSLSINDGFHVY...  \n",
       "1  P36896[24-126]  SGPRGVQALLCACTSCLQANYTCETDGACMVSIFNLDGMEHHVRTC...  \n",
       "2  Q8NER5[22-113]  LSPGLKCVCLLCDSSNFTCQTEGACWASVMLTNGKEQVIKSCVSLP...  \n",
       "3  Q13705[19-137]  SGRGEAETRECIYYNANWELERTNQSGLERCEGEQDKRLHCYASWR...  \n",
       "4  P37023[22-118]  DPVKPSRGPLVTCTCESPHCKGPTCRGAWCTVVLVREEGRHPQEHR...  "
      ]
     },
     "execution_count": 133,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "together['Reactome'].fillna(together['X_talk pathway'], inplace=True)\n",
    "del together['X_talk pathway']\n",
    "together.head(1)\n",
    "print(together.shape)\n",
    "together.head(5)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 136,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Entry</th>\n",
       "      <th>positionID</th>\n",
       "      <th>Gene names</th>\n",
       "      <th>Protein names</th>\n",
       "      <th>sequences</th>\n",
       "      <th>Seq_length</th>\n",
       "      <th>Protein Expression : High</th>\n",
       "      <th>Protein Expression : Med</th>\n",
       "      <th>Protein Expression : Low</th>\n",
       "      <th>Reactome</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>Q04771</td>\n",
       "      <td>Q04771[21-123]</td>\n",
       "      <td>ACVR1 ACVRLK2</td>\n",
       "      <td>Activin receptor type-1 (EC 2.7.11.30) (Activi...</td>\n",
       "      <td>MEDEKPKVNPKLYMCVCEGLSCGNEDHCEGQQCFSSLSINDGFHVY...</td>\n",
       "      <td>103</td>\n",
       "      <td>glandular cells, smooth muscle cells</td>\n",
       "      <td>myocytes, glandular cells, glandular cells, gl...</td>\n",
       "      <td>squamous epithelial cells, glial cells, pneumo...</td>\n",
       "      <td>TGF(core),TGF(core)</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "    Entry      positionID     Gene names  \\\n",
       "0  Q04771  Q04771[21-123]  ACVR1 ACVRLK2   \n",
       "\n",
       "                                       Protein names  \\\n",
       "0  Activin receptor type-1 (EC 2.7.11.30) (Activi...   \n",
       "\n",
       "                                           sequences  Seq_length  \\\n",
       "0  MEDEKPKVNPKLYMCVCEGLSCGNEDHCEGQQCFSSLSINDGFHVY...         103   \n",
       "\n",
       "              Protein Expression : High  \\\n",
       "0  glandular cells, smooth muscle cells   \n",
       "\n",
       "                            Protein Expression : Med  \\\n",
       "0  myocytes, glandular cells, glandular cells, gl...   \n",
       "\n",
       "                            Protein Expression : Low             Reactome  \n",
       "0  squamous epithelial cells, glial cells, pneumo...  TGF(core),TGF(core)  "
      ]
     },
     "execution_count": 136,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "together.drop_duplicates(keep='first')\n",
    "together = together[['Entry','positionID','Gene names','Protein names','sequences','Seq_length','Protein Expression : High','Protein Expression : Med','Protein Expression : Low','Reactome']]\n",
    "together.to_csv(r\"519together_pathway.csv\")\n",
    "together.head(1)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 117,
   "metadata": {},
   "outputs": [],
   "source": [
    "together.to_csv(r\"together_pathway.csv\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.4"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
